crates/ruma-client/Cargo.toml

```diff
@@ -37,18 +36,14 @@ assign = { workspace = true }
 async-stream = "0.3.0"
 bytes = "1.0.1"
 futures-core = "0.3.8"
-futures-lite = { version = "1.11.3", optional = true }
 http = { workspace = true }
 hyper = { version = "0.14.2", optional = true, features = ["client", "http1", "http2", "tcp"] }
 hyper-rustls = { version = "0.24.0", optional = true, default-features = false }
 hyper-tls = { version = "0.5.0", optional = true }
-isahc = { version = "1.3.1", optional = true }
 reqwest = { version = "0.11.4", optional = true, default-features = false }
 ruma-client-api = { workspace = true, optional = true, features = ["client"] }
 ruma-common = { workspace = true, features = ["api"] }
-serde = { workspace = true }
 serde_html_form = { workspace = true }
-serde_json = { workspace = true }
```

Comment:
So these two serde dependencies were unused? This cleanup should be in its own commit.

Reply:
Yes, those two were unused. I separated them into own commit.